src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -2582,6 +2582,8 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
     {
         if (!_closing.exchange(true))
         {
+            _RestorePointerCursorHandlers(*this, nullptr);

```

Comment:
hmm.. will this restore the cursor to pointer if a terminal closes _while a different one is the one that hid it_? The `sleep 5; exit` scenario, but you go type in a different terminal tab while it's waiting to beef it

Reply:
Once again this user who wants to destroy terminal? :smile: 
So the restore is managed the TerminalPage level so we won't need to remember who hid it. 
If I don't miss anything In the worst case we will get mouse restored..

Reply:
Let's leave it like this. This will be okay for now  :smile:

Reply:
> user who wants to destroy terminal

You'd be surprised! Look for posts by `vefatica` here :wink: and there's also one Microsoft person who keeps reporting bugs in multibyte character handling because he apparently just runs a stress testing program that emits random code units and then he actively resizes the window while it's doing that

Reply:
> You'd be surprised! Look for posts by `vefatica` here 😉 and there's also one Microsoft person who keeps reporting bugs in multibyte character handling because he apparently just runs a stress testing program that emits random code units and then he actively resizes the window while it's doing that

After few months following the tracker I am not surprised with anything as the creativity is limitless :blush:  And I am a big fan of vefatica quite for a  while - even before he found the handle leak.